### ISSUE TITLE
Squiz.ControlStructures.ControlSignature does not handle alt syntax when checking space after closing brace

### ISSUE THREAD

**Issue by LC43** [squizlabs/PHP_CodeSniffer#1387]
Hi, thanks for your work! I have a question, which I think is a bug:

closing a brace inside a another if that uses the alternative syntax:

```php
$a = 1;
if (1 === $a) :
    if ($b) {
        echo 'b';
    } else {
        $something = 'not b';
    }
else :
    echo 'not 1';
endif;
```
Error:
```
$ phpcs --report-full --standard=PSR2 test-alternate.php
----------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------
 8 | ERROR | [x] Expected 1 space after closing brace; newline found
----------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------
```
using this code, i found that phpcs fails on the line 8 `    }`. It shouldn't consider the next line as the continuation of the inner control structure.

adding a new line between `}` and `else :` removes the error message:

```php
    }
    $b = 1;
else :
```

I inherit the code, so I just replaced the syntax. But on other occasions, should I follow the advice on #347 and use custom ruleset or is this a bug?

this is a gist with both blocks. It fails on the first and passes the second:

https://gist.github.com/LC43/383ff51e169b9b113cdb750dbcdcf25b

thank you.

**Comment by gsherwood** [comment]
This looks like a bug in the sniff to me. The core tokenizer has processed this code structure correctly, but the sniff is confused by the fact that `else` is both the closing point for the `if` and the opening point for itself.

**Comment by gsherwood** [comment]
This has been fixed now. There wont be any errors reported for that code when using PSR2. Thanks for reporting this.

**Comment by LC43** [comment]
wow! impressive @gsherwood, thank you so much for your fast fix!
